commands/moderation/filter.js

```diff
@@ -0,0 +1,113 @@
+/*  
+  List of offensive words for easy readability
+  
+    Strong Words 
+      *one of these words will trigger the filter 
+       hell          damn
+       goddamn       goddamned
+       goddamnit     damned
+       damnit        hell
+       fuck          fucked
+       fucking       fucker
+       fuckwad       buttplug
+       motherfucker  wtf
+       nigger        nigga
+       penis         vagina
+       asshole       sex
+       sexed         piss
+       pissed        sexy
+       sexual        sexuality
+       bastard       clitoris
+       bitch         boobs
+       semen         sperm
+       jizzed        jizz
+       whore         prostitute
+       fornicate     fornication
+       adultery      adulter
+       adulteress    slut
+       condom
+       
+    Minor Words    
+       *3+ of these words will trigger the filter
+       darn        dern
+       darned      derned
+       darnit      dernit
+       dang        heck
+       frick       fricked
+       crap        omg
+      
+*/
+
+const Discord = require('discord.js');
+
+module.exports = {
+  name: "filter",
+  description: "filter a message",
+
+  execute(msg) {
+    // Make sure user should be filtered
+    if (
+    !msg.member.roles.cache.some(
+      (role) =>
+        role.name === 'Super User' ||
+        role.name === 'Moderator' ||
+        role.name === 'Admin'
+    )
+  ) {
+      // list of words in users message
+      const words = msg.content.toLowerCase().split(' ');
+
+      // regex expression of strong offensive words
+      const strongWords = /hell|damn|goddamn|goddamned|damned|damnit|hell|fuck|fucked|fucking|fucker|fuckwad|fuckwad|motherfucker|wtf|nigger|nigga|penis|vagina|asshole|sex|sexed|piss|pissed|sexy|sexual|sexuality|bastard|bitch|boobs|semen|sperm|jizz|jizzed|whore|prostitute|fornicate|fornication|adultery|adulter|adulteress|slut|buttplug|clitoris|condom[.,/\\:;\*\?\!\@\#\$]?/;
+
+      // regex expression of minor offensive words
+      const lightWords = /darn|dern|darned|derned|darnit|dernit|dang|heck|frick|fricked|crap|omg[.,/\\:;\*\?\!\@\#\$]?/;
```

Comment:
I don't think we really need the punctuation at the end (`[.,/\\:;\*\?\!\@\#\$]?`) since `RegExp.test()` will still match the word whether it's got punctuation at the end or not.

Let's remove it unless there's a specific reason we need it.

Reply:
Did not know this! Done